docs: add `rover license fetch` command (#1795)
apollographql/apollo-cli: docs/source/commands/license.mdx

---
title: The Rover license command
description: Manage Enterprise licenses 
---

Rover commands that manage Enterprise licenses for Apollo begin with `rover license`.

## Fetch offline Enterprise license 

> Fetch an offline Enterprise license from GraphOS with `rover license fetch`. Specify the graph with a graph ref argument: 
> 
> ```bash
> rover license fetch <apollo-graph-ref>
> ```

Reviewer: ```suggestion
Fetch an offline Enterprise license from GraphOS with `rover license fetch`. Specify the graph with a graph id argument: 

```bash
rover license fetch --graph-id <apollo-graph-id>
```
```

---
title: The Rover license command
description: Manage Enterprise licenses 

> ---

Reviewer 2: Can we add the minVersion to the frontmatter?

> Rover commands that manage Enterprise licenses for Apollo begin with `rover license`.

Reviewer 2: Other than fetching offline licenses, there isn't any other use for the `license fetch` command, right?

As a reader, I might see this page/intro sentence and wonder if I'm supposed to be using this command for license management. It might be good to add a note that regular (i.e. non-offline) licenses are automatically fetched via Uplink, so there's no need to run this command for them.

## Fetch offline Enterprise license 

### `license fetch`

> Run `rover license fetch` to fetch a license from GraphOS for a graph with a GraphOS Enterprise plan.

Reviewer 2: ```suggestion
Run `rover license fetch` to fetch an offline license from GraphOS for a graph with a GraphOS Enterprise plan.
```

Specify the graph to fetch from with a `--graph-id` argument: 

```bash
rover license fetch --graph-id <apollo-graph-id>
```

<Note>

* An offline Enterprise license is valid for the lesser of the duration of your contract with Apollo, or one year, with an added grace period of 28 days. 

> * You are responsible for keeping your offline license files up to date within your infrastructure.

Reviewer 2: I couldn't find anything about "keeping your offline license files up to date within your infrastructure" in https://github.com/apollographql/router/pull/4327/files ... Does this phrase just mean running the `rover license fetch` command at least once a year? If so, maybe something like this for clarity:

```suggestion
* You are responsible for keeping your offline license files up to date within your infrastructure by running `rover license fetch` at least once a year.
```

Also, is offline license expiration something that's worth mentioning in the router docs as well?

---
title: The Rover license command
description: Manage GraphOS Enterprise licenses, including offline licenses, with the rover license command 
minVersion: 0.22.0
---

Rover commands that manage Enterprise licenses for Apollo begin with `rover license`.

## Fetch offline Enterprise license 

> ### `license fetch`

Reviewer: Let's call out that you need to be enabled to fetch offline licenses.
